sdk/appconfiguration/azure-appconfiguration-provider/tests/test_setting_selector.py

```diff
@@ -0,0 +1,93 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import pytest
+from azure.appconfiguration.provider._models import SettingSelector
+from azure.appconfiguration.provider._constants import EMPTY_LABEL
+
+
+class TestSettingSelector:
+    """Tests for the SettingSelector class."""
+
+    def test_setting_selector_init(self):
+        """Test the initialization of SettingSelector with valid parameters."""
+        # Basic initialization with key_filter only
+        selector = SettingSelector(key_filter="*")
+        assert selector.key_filter == "*"
+        assert selector.label_filter == EMPTY_LABEL
+        assert selector.tag_filters is None
+
+        # Initialization with all parameters
+        selector = SettingSelector(
+            key_filter="my-key*", label_filter="my-label", tag_filters=["tag1=value1", "tag2=value2"]
+        )
+        assert selector.key_filter == "my-key*"
+        assert selector.label_filter == "my-label"
+        assert selector.tag_filters == ["tag1=value1", "tag2=value2"]
+
+    def test_setting_selector_tag_filters_validation(self):
+        """Test the validation of tag_filters in SettingSelector."""
+        # Valid tag filters
+        selector = SettingSelector(key_filter="*", tag_filters=["tag=value"])
+        assert selector.tag_filters == ["tag=value"]
+
+        selector = SettingSelector(key_filter="*", tag_filters=["tag=value", "another=one"])
+        assert selector.tag_filters == ["tag=value", "another=one"]
+
+        selector = SettingSelector(key_filter="*", tag_filters=["tag="])
+        assert selector.tag_filters == ["tag="]
```

Comment:
Can you add test for tag with null value

Reply:
Is this meant to be a passing test? I can't get it to work, I get a bad request when creating it.

Reply:
Yes, this is a valid input. SDK should support `\0` as tag filter value and URL encode it as `%00` before making the request to AppConfig backend.